External/FEXCore/include/FEXCore/HLE/SyscallHandler.h

```diff
@@ -43,6 +49,24 @@ namespace FEXCore::HLE {
     OS_HANGOVER,
   };
 
+  class SyscallHandler;
+  struct AOTIRCacheEntryLookupResult {
```

Comment:
This critically needs a name that conveys it's actually holding a lock. The interface is a disaster waiting to happen if such a name can't be found, and I'd much rather we drop it in that case and fall back to more verbose code instead.

Reply:
Uh, I don't quite agree it's that bad, it is used from one or two places, and even if you don't know it is holding the lock it does work as one expects it to.

I'm up for any better name recommendations though.

Reply:
I think this is best tracked with a docstring in `LookupAOTIRCacheEntry`

Reply:
(added docstring to LookupAOTIRCacheEntry)